src/Geolocation/Geolocation.php

```diff
@@ -0,0 +1,336 @@
+		if (isset($_COOKIE[self::GEOLOCATION_COOKIE])) {
+			return;
+		}
+
+		setcookie(
+			self::GEOLOCATION_COOKIE,
+			$this->getLocationDetails(),
+			time() + DAY_IN_SECONDS,
+			'/'
+		);
+	}
+
+	/**
+	 * Check if user location exists in the provided locations.
+	 *
+	 * @param string $formId Form Id.
+	 * @param array<string, mixed> $defaultLocations Default locations set to form..
+	 * @param array<string, mixed> $additionalLocations Additional location set to form.
+	 *
+	 * @return string
+	 */
+	public function isUserGeolocated(string $formId, array $defaultLocations, array $additionalLocations): string
+	{
+		// Add ability to disable geolocation from external source. (Generaly used for GDPR).
+		$filterName = Filters::getGeolocationFilterName('disable');
+		if (has_filter($filterName) && apply_filters($filterName, false) === true) {
+			Helper::logger([
+				'geolocation' => 'Filter disabled active, skip geolocation.',
+				'formIdOriginal' => $formId,
+				'formIdUsed' => $formId,
+				'userLocation' => '',
+			]);
+			return $formId;
+		}
+
+		// Returns user location got from the API or Cookie.
+		$userLocation = $this->getLocationDetails();
+
+		// Check if additional location exists on the form.
+		if ($additionalLocations) {
+			$matchAdditionalLocations = [];
+
+			// Iterate all additional location to find the first that match.
+			foreach ($additionalLocations as $additionalLocation) {
+				// Find geolocation from array of options.
+				$geoLocation = array_filter(
+					$additionalLocation['geoLocation'] ?? [],
+					function ($geo) use ($userLocation) {
+						$country = $this->getCountryGroup($geo['value']);
+
+						return isset($country[$userLocation]) ?? false;
+					}
+				) ?? [];
+
+				// Exit after first sucesfull result.
+				if ($geoLocation) {
+					$matchAdditionalLocations = $additionalLocation;
+					break;
+				}
+			}
+
+			// If additional locations match output that new form.
+			if ($matchAdditionalLocations) {
+				Helper::logger([
+					'geolocation' => 'Locations exists, locations match. Outputing new form.',
+					'formIdOriginal' => $formId,
+					'formIdUsed' => $matchAdditionalLocations['formId'] ?? '',
+					'userLocation' => $userLocation,
+				]);
+				return $matchAdditionalLocations['formId'] ?? '';
+			}
+		}
+
+		// If thare are not location but we have default locations set match that array with the user location.
+		if ($defaultLocations) {
+			$matchDefaultLocations = array_filter(
+				$defaultLocations,
+				function ($location) use ($userLocation) {
+					$country = $this->getCountryGroup($location['value']);
+
+					return isset($country[$userLocation]) ?? false;
+				}
+			);
+
+			// Return first result.
+			$matchDefaultLocations = reset($matchDefaultLocations);
+
+			// If default locations match output that new form.
+			if ($matchDefaultLocations) {
+				Helper::logger([
+					'geolocation' => 'Locations doesn\'t match or exist, default location match. Outputing new form.',
```

Comment:
```suggestion
					'geolocation' => 'Selected location doesn\'t match the default location (or it doesn'\t exist). Outputing new form.',
```

Not 100% sure what was meant in the original message, is my suggestion a correct assumption?